src/dhcp6relay/src/relay.cpp

```diff
@@ -378,44 +378,57 @@ void prepare_socket(int *local_sock, int *server_sock, relay_config *config, int
     memset(&ll_addr, 0, sizeof(ll_addr));
 
     if ((*local_sock = socket(AF_INET6, SOCK_DGRAM, 0)) == -1) {
-        syslog(LOG_ERR, "socket: Failed to create socket\n");
+        syslog(LOG_ERR, "socket: Failed to create socket on interface %s\n", config->interface.c_str());
     }
 
     if ((*server_sock= socket(AF_INET6, SOCK_DGRAM, 0)) == -1) {
-        syslog(LOG_ERR, "socket: Failed to create socket\n");
+        syslog(LOG_ERR, "socket: Failed to create socket on interface %s\n", config->interface.c_str());
     }
 
+    int retry = 0;
+    int bind_addr = 0;
+    int bind_ll_addr = 0;
+    do {
+        if (getifaddrs(&ifa) == -1) {
+	    syslog(LOG_WARNING, "getifaddrs: Unable to get network interfaces\n");
+	    exit(1);
+	}
 
-    if (getifaddrs(&ifa) == -1) {
-        syslog(LOG_WARNING, "getifaddrs: Unable to get network interfaces\n");
-        exit(1);
-    }
+	ifa_tmp = ifa;
+	while (ifa_tmp) {
+	    if (ifa_tmp->ifa_addr->sa_family == AF_INET6) {
+	        struct sockaddr_in6 *in6 = (struct sockaddr_in6*) ifa_tmp->ifa_addr;
+		if((strcmp(ifa_tmp->ifa_name, config->interface.c_str()) == 0) && !IN6_IS_ADDR_LINKLOCAL(&in6->sin6_addr)) {  
```

Comment:
Looks like `strcmp(ifa_tmp->ifa_name, config->interface.c_str())` is happening twice (once here and once in the next if-check). Restructure this section so that only one string comparison is done.